LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -143,73 +144,83 @@ void AddToModuleInstructorList(Instructor instructor)
+
+                    if (!moduleIdSet.Contains(moduleId))
+                    {
+                        moduleIdSet.Add(moduleId);
+                        Module module;
+                        moduleDictionary.TryGetValue(moduleId, out module);
+                        moduleList.Add(module);
+                    }
+                    i++;
                 }
-                i++;
             }
             return moduleList;
         }
 
         public void CreateClassrooms(int classroomCount)
         {
-            Guid moduleId = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
-            Module module;
-            moduleDictionary.TryGetValue(moduleId, out module);
-            int i = 0;
-            while (i < classroomCount)
+            if(moduleIdList.Count != 0)
             {
-                Classroom classroom = ObjectGenerator.GenerateClassroom(module);
-                classroomList.Add(classroom);
-                i++;
-            }
+                ReportHeader("Create Classroom");
+                Guid moduleId = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
+                Module module;
+                moduleDictionary.TryGetValue(moduleId, out module);
+                int i = 0;
+                while (i < classroomCount)
+                {
+                    Classroom classroom = ObjectGenerator.GenerateClassroom(module);
+                    ReportItem(classroom.ToString(), "Created classroom", i++);
+                    classroomList.Add(classroom);
+                    i++;
+                }
+                ReportFooter("Create Classroom");
+            } 
         }
 
         public void AssignInstructorsToClassrooms()
         {
-            foreach (var classroom in classroomList)
+            if(classroomList.Count != 0)
             {
-                if (moduleInstructorDictionary.ContainsKey(classroom.Module.Id))
+                foreach (var classroom in classroomList)
                 {
-                    List<Instructor> instructorList;
-                    moduleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
-                    classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
-                }
-                else
-                {
-                    Console.WriteLine("Failed to find instructor for classroom");
+                    if (moduleInstructorDictionary.ContainsKey(classroom.Module.Id))
+                    {
+                        List<Instructor> instructorList;
+                        moduleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
+                        classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
+                    }
+                    else
+                    {
+                        Console.WriteLine("Failed to find instructor for classroom");
```

Comment:
Create a method to report an error: ReportError(string errorDescription). Use it to report:
Console.WriteLine("Failed to find instructor for classroom");